packages/glimmer-runtime/lib/dom/helper.ts

```diff
@@ -165,16 +203,77 @@ class DOMHelper {
   }
 }
 
+function defaultInsertHTMLBefore(this: void, useless: HTMLElement, _parent: Element, nextSibling: Node, html: string): Bounds { // tslint:disable-line
+  // TypeScript vendored an old version of the DOM spec where `insertAdjacentHTML`
+  // only exists on `HTMLElement` but not on `Element`. We actually work with the
+  // newer version of the DOM API here (and monkey-patch this method in `./compat`
+  // when we detect older browsers). This is a hack to work around this limitation.
+  let parent = _parent as HTMLElement;
+
+  let prev = nextSibling ? nextSibling.previousSibling : parent.lastChild;
+  let last;
+
+  if (html === null || html === '') {
+    return new ConcreteBounds(parent, null, null);
+  }
+
+  if (nextSibling === null) {
+    parent.insertAdjacentHTML('beforeEnd', html);
+    last = parent.lastChild;
+  } else if (nextSibling instanceof HTMLElement) {
+    nextSibling.insertAdjacentHTML('beforeBegin', html);
+    last = nextSibling.previousSibling;
+  } else {
+    // Non-element nodes do not support insertAdjacentHTML, so add an
+    // element and call it on that element. Then remove the element.
+    //
+    // This also protects Edge, IE and Firefox w/o the inspector open
+    // from merging adjacent text nodes. See ./compat/text-node-merging-fix.ts
+    parent.insertBefore(useless, nextSibling);
+    useless.insertAdjacentHTML('beforeBegin', html);
+    last = useless.previousSibling;
+    parent.removeChild(useless);
+  }
+
+  let first = prev ? prev.nextSibling : parent.firstChild;
+  return new ConcreteBounds(parent, first, last);
+}
+
+function fixSVG(this: void, useless: HTMLElement, _parent: Element, nextSibling: Node, html: string): Bounds { // tslint:disable-line
+  // IE, Edge: also do not correctly support using `innerHTML` on SVG
+  // namespaced elements. So here a wrapper is used.
+  let wrappedHtml = '<svg>' + html + '</svg>';
+
+  useless.innerHTML = wrappedHtml;
+
+  let [first, last] = moveNodesBefore(useless.firstChild, _parent, nextSibling);
+  return new ConcreteBounds(_parent, first, last);
+}
+
+export function insertHTMLBefore(this: void, useless: HTMLElement, _parent: Element, nextSibling: Node, html: string): Bounds { // tslint:disable-line
```

Comment:
@wycats We need to compose together the fixes we were doing in lines [here](https://github.com/tildeio/glimmer/pull/238/files#diff-b68fc8c7a24a4446b4a4660993e66827R272)